samples/msgext-northwind-inventory-csharp/MessageExtensions/productSearchCommand.cs

```diff
@@ -0,0 +1,95 @@
+            {
+                var values = query.Parameters[0]?.Value.ToString().Split(',');
+                productName = CleanupParam(values.ElementAtOrDefault(0));
+                categoryName = CleanupParam(values.ElementAtOrDefault(1));
+                inventoryStatus = CleanupParam(values.ElementAtOrDefault(2));
+                supplierCity = CleanupParam(values.ElementAtOrDefault(3));
+                stockLevel = CleanupParam(values.ElementAtOrDefault(4));
+            }
+            else
+            {
+                productName = CleanupParam(query.Parameters?.FirstOrDefault(p => p.Name == "productName")?.Value as string);
+                categoryName = CleanupParam(query.Parameters?.FirstOrDefault(p => p.Name == "categoryName")?.Value as string);
+                inventoryStatus = CleanupParam(query.Parameters?.FirstOrDefault(p => p.Name == "inventoryStatus")?.Value as string);
+                supplierCity = CleanupParam(query.Parameters?.FirstOrDefault(p => p.Name == "supplierCity")?.Value as string);
+                stockLevel = CleanupParam(query.Parameters?.FirstOrDefault(p => p.Name == "stockQuery")?.Value as string);
+            }
+
+            Console.WriteLine($"🔎 Query #{++queryCount}:\nproductName={productName}, categoryName={categoryName}, inventoryStatus={inventoryStatus}, supplierCity={supplierCity}, stockLevel={stockLevel}");
+
+            ProductService productService = new ProductService(configuration);
+            var products = await productService.SearchProducts(productName, categoryName, inventoryStatus, supplierCity, stockLevel);
+
+            Console.WriteLine($"Found {products.Count} products in the Northwind database");
+            var attachments = new List<MessagingExtensionAttachment>();
+
+            foreach (var product in products)
+            {
+                var preview = new HeroCard
+                {
+                    Title = product.ProductName,
+                    Subtitle = $"Supplied by {product.SupplierName} of {product.SupplierCity}<br />{product.UnitsInStock} in stock",
+                    Images = new List<CardImage> { new CardImage(product.ImageUrl) }
+                }.ToAttachment();
+
+                var resultCard = CardHandler.GetEditCard(product);
+
+                var attachment = new MessagingExtensionAttachment
+                {
+                    ContentType = resultCard.ContentType,
+                    Content = resultCard.Content,
+                    Preview = preview
+                };
+
+                attachments.Add(attachment);
+            }
+
+            return new MessagingExtensionResponse
+            {
+                ComposeExtension = new MessagingExtensionResult
+                {
+                    Type = "result",
+                    AttachmentLayout = "list",
+                    Attachments = attachments
+                }
+            };
+        }
+
+        private static string CleanupParam(string value)
```

Comment:
if we are using this in multiple places then we can put into common class and use every places.

Reply:
fixed by adding common method